### ISSUE TITLE
12$: improve error for '<test>{ 4 + 3</test>'

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#732]
From: https://github.com/rome/tools/issues/2625#issuecomment-1140239006
```js
<test>{ 4 + 3</test>
```
actual:
```
hello.js:1:15: error: unclosed regexp literal [E0039]
hello.js:2:1: error: unexpected token [E0054]
```
expected:
```
hello.js:1:14: error: expected '}' to close JSX expression thingy
hello.js:1:7: note: JSX expression thingy opened here
```

**Comment by vegerot** [comment]
I think we should start adding snarky examples for these issues too

**Comment by vegerot** [comment]
What would be an example snarky message to give the user in this case?
